cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -291,53 +287,25 @@ private void stopVpnTunnel() {
     });
   }
 
-  /* Checks that the remote server is reachable, allows UDP forwarding, and the credentials are
-   * valid. Executes the three checks in parallel in order to minimize the user's wait time. */
-  private OutlinePlugin.ErrorCode checkServerConnectivity(final String localServerAddress,
-      final int localServerPort, final String remoteServerAddress, final int remoteServerPort) {
-    final Callable<Boolean> udpForwardingCheck = new Callable<Boolean>() {
-      public Boolean call() {
-        return ShadowsocksConnectivity.isUdpForwardingEnabled(localServerAddress, localServerPort);
-      }
-    };
-    final Callable<Boolean> reachabilityCheck = new Callable<Boolean>() {
-      public Boolean call() {
-        return ShadowsocksConnectivity.isServerReachable(remoteServerAddress, remoteServerPort);
-      }
-    };
-    final Callable<Boolean> credentialsValidationCheck = new Callable<Boolean>() {
-      public Boolean call() {
-        return ShadowsocksConnectivity.validateServerCredentials(
-            localServerAddress, localServerPort);
-      }
-    };
+  private OutlinePlugin.ErrorCode checkServerConnectivity(final JSONObject serverConfig) {
     try {
-      Future<Boolean> udpCheckResult = executorService.submit(udpForwardingCheck);
-      Future<Boolean> reachabilityCheckResult = executorService.submit(reachabilityCheck);
-      Future<Boolean> credentialsCheckResult = executorService.submit(credentialsValidationCheck);
-      boolean isUdpForwardingEnabled = udpCheckResult.get();
-      if (isUdpForwardingEnabled) {
-        // The UDP forwarding check is a superset of the TCP checks. Don't wait for the other tests
-        // to complete; if they fail, assume it's due to intermittent network conditions and declare
-        // success anyway.
+      long errorCode = shadowsocks.Shadowsocks.checkConnectivity(serverConfig.getString("host"),
+          serverConfig.getInt("port"), serverConfig.getString("password"),
+          serverConfig.getString("method"));
+      LOG.info(String.format(Locale.ROOT, "Go connectivity check result:  %d", errorCode));
+      if (errorCode == OutlinePlugin.ErrorCode.NO_ERROR.value) {
```

Comment:
Do you need to filter the value here?  This is platform-specific code, so only platform-specific logic should occur here.  What happens if you just `return errorCode` without inspecting it?

Reply:
The method returns `OutlinePlugin.ErrorCode` because the rest of the code operates with this type. I'm happy to refactor the class to work with int values and move this logic to `OutlinePlugin`if you believe it is clearer.

Reply:
OK.  You could `return OutlinePlugin.ErrorCode.values()[errorCode]`, but I don't have a strong preference.

Ultimately, I think we should fix this so we don't have five copies of the error code enum, but we don't have to do that in this PR.

Reply:
Done. 

> Ultimately, I think we should fix this so we don't have five copies of the error code enum, but we don't have to do that in this PR.

I agree, but it's not entirely clear to me how to achieve it accross language boundaries.